src/build_test.cc

```diff
@@ -959,6 +960,46 @@ TEST_F(BuildTest, OrderOnlyDeps) {
   ASSERT_EQ(1u, command_runner_.commands_ran_.size());
 }
 
+TEST_F(BuildTest, DepsReferencedThroughParentFolder) {
+  string foo_depfile_content = "foo.o: ../project/blah.h\n";
+
+  string err;
+  ASSERT_NO_FATAL_FAILURE(AssertParse(&state_,
+"rule cc\n  command = cc $in\n  depfile = $out.d\n"
+"rule touch\n  command = touch $out\n"
+"build foo.o: cc foo.c || blah.h\n"
+"build blah.h: touch blah.stamp\n"));
+  SetWorkingDirForCanonicalizePath("/root/project");
```

Comment:
Removing this line with `SetWorkingDirForCanonicalizePath()` will fail the test.